### PR TITLE
🎉 Add Airbyte-native shopify connector (replaces Singer-based)

### LINKED ISSUE #3581: 🎉 new source: Airbyte-native Shopify connector
## Tell us about the new integration you’d like to have
Our current shopify connector is based on Singer. This makes the iteration cycle longer as described in #3582 . We'd like to migrate our Shopify source connector off of the Singer-based one and onto an Airbyte-based package using the CDK. 

## Describe the alternative you are considering or using

┆Issue is synchronized with this [Asana task](https://app.asana.com/0/1200368060087111/1200376844471650) by [Unito](https://www.unito.io)

### PR DESCRIPTION
**Pull request by bazarnov**
## What
#3581 new source: Airbyte-native Shopify connector

## How
Using Airbyte's Python CDK, implemented the logic for source-shopify connector, with the same endpoint as shopify-singer connector has.
By default, it pulls out all the data, then keeps it fresh using the incremental refresh (incremental should be selected as an option inside connection settings)

## Pre-merge Checklist
Expand the checklist which is relevant for this PR. 

<details><summary> <strong> Connector checklist </strong> </summary>
<p>

- [x] Issue acceptance criteria met
- [x] Unit & integration tests added as appropriate (and are passing)
- [x] `/test` command documented [here]() is passing. 
- [x] Credentials added to Github CI if needed and not already present. [instructions for injecting secrets into CI](https://docs.airbyte.io/contributing-to-airbyte/building-new-connector#using-credentials-in-ci). 
- [x] Code reviews completed
- [x] Documentation updated 
    - [x] README
    - [x] Reference docs in the `docs/integrations/` directory.
- [x] Build is successful
- [x] Connector version bumped like described [here](https://docs.airbyte.io/contributing-to-airbyte/building-new-connector#updating-a-connector)
- [x] New Connector version released on Dockerhub by running the `/publish` command described [here](https://docs.airbyte.io/contributing-to-airbyte/building-new-connector#updating-a-connector)
- [x] No major blockers
- [ ] PR merged into master branch
- [ ] Follow up tickets have been created
- [ ] Associated tickets have been closed & stakeholders notified
</p>
</details>

### CODE REVIEW

File: airbyte-integrations/connectors/source-shopify/source_shopify/source.py
Comment (sherifnada) [inline airbyte-integrations/connectors/source-shopify/source_shopify/source.py:41]: this can be `https://{self.api_key}:{self.api_password}@{self.shop}.myshopify.com/admin/api/` and then the paths in subclasses can just be the `{self.api_version}....` portion?

File: airbyte-integrations/connectors/source-shopify/source_shopify/source.py
Comment (sherifnada) [inline airbyte-integrations/connectors/source-shopify/source_shopify/source.py:59]: why is there a sleep? can you leave a comment/

File: airbyte-integrations/connectors/source-shopify/source_shopify/source.py
Comment (sherifnada) [reply airbyte-integrations/connectors/source-shopify/source_shopify/source.py:41]: also I think we can just use the `Authorization` header like described in the [docs](https://shopify.dev/tutorials/authenticate-a-private-app-with-shopify-admin#make-authenticated-requests): 

> If your HTTP client doesn't support basic authentication using this method, then you can provide the credentials in the Authorization header field instead:

File: airbyte-integrations/connectors/source-shopify/source_shopify/source.py
Comment (sherifnada) [inline airbyte-integrations/connectors/source-shopify/source_shopify/source.py:53]: how come we are not using the `rel` header from the response to find the next page? https://shopify.dev/tutorials/make-paginated-requests-to-rest-admin-api#how-it-works

File: airbyte-integrations/connectors/source-shopify/source_shopify/source.py
Comment (bazarnov) [reply airbyte-integrations/connectors/source-shopify/source_shopify/source.py]: This was only for the "slow" debugging" using console output, it's not going to be a part of the code.

File: airbyte-integrations/connectors/source-shopify/source_shopify/source.py
Comment (bazarnov) [reply airbyte-integrations/connectors/source-shopify/source_shopify/source.py:38]: Basically that's the same, I used, but I didn't use base64 in order to encode the base URL. I'll try to implement this, thanks.

File: airbyte-integrations/connectors/source-shopify/source_shopify/source.py
Comment (bazarnov) [reply airbyte-integrations/connectors/source-shopify/source_shopify/source.py:50]: You're right, there is the opportunity to overcome complex pagination with this, for now it work as expected, but this defenetely should be easier, so I'll work on in

File: airbyte-integrations/connectors/source-shopify/source_shopify/source.py
Comment (sherifnada) [inline airbyte-integrations/connectors/source-shopify/source_shopify/source.py:100]: is the cursor field always guaranteed to be an int? if so then I believe this looks right. Though you may want to do 

```
{
  self.cursor_field: max(
    latest_record.get(self.cursor_field, 0), # is the record guaranteed to have the cursor field populated? 
    current_stream_state.get(self.cursor_field, 0)
  )
}
```

File: airbyte-integrations/connectors/source-shopify/source_shopify/source.py
Comment (sherifnada) [inline airbyte-integrations/connectors/source-shopify/source_shopify/source.py:175]: if `id` is always the cursor field then we can probably just hardcode it at the incremental stream base class level. If a subclass needs to change it, then it can override the `cursor_field` property.

File: airbyte-integrations/connectors/source-shopify/source_shopify/source.py
Comment (sherifnada) [inline airbyte-integrations/connectors/source-shopify/source_shopify/source.py:105]: why don't we use `updated_at` as the cursor field instead of ID? It's easier to understand. It seems like `Customer` and `AbandonedCheckouts` at least support this -- i haven't verified all endpoints but worth checking

File: airbyte-integrations/connectors/source-shopify/source_shopify/source.py
Comment (sherifnada) [inline airbyte-integrations/connectors/source-shopify/source_shopify/source.py:87]: This incremental implementation looks sound! small comment but otherwise I think it LGTM

File: airbyte-integrations/connectors/source-shopify/source_shopify/source.py
Comment (bazarnov) [reply airbyte-integrations/connectors/source-shopify/source_shopify/source.py:100]: Yes, the id is always an "int". I agree to add the default state as 0, probably missed that, my bad.

File: airbyte-integrations/connectors/source-shopify/source_shopify/source.py
Comment (bazarnov) [reply airbyte-integrations/connectors/source-shopify/source_shopify/source.py:105]: Based on Shopify's API Reference:
Ref link: [link](https://shopify.dev/docs/admin-api/rest/reference/orders/order)
"Filtering orders by updated_at is not an effective method for fetching orders because its value can change when no visible fields of an order have been updated. Use the Webhook and Event APIs to subscribe to order events instead."
They recommend using "since_id" because id is always "int", and it always += 1 for everything. Thus you can get the list of id's and sort them DESC, you'll get the latest first. So decided to use this. But probably, you've got the point here.

File: airbyte-integrations/connectors/source-shopify/source_shopify/source.py
Comment (bazarnov) [reply airbyte-integrations/connectors/source-shopify/source_shopify/source.py:55]: Manage to achieve this only by changing the @property of the HttpStream class, like above, since when I use simple hardcode a.k.a url_base = "..." it throws the error "value is not-accessible". Is it bad to use it by changing the property of the class like os?

File: airbyte-integrations/connectors/source-shopify/source_shopify/source.py
Comment (bazarnov) [reply airbyte-integrations/connectors/source-shopify/source_shopify/source.py:90]: Same as with url_base.

**Review by yaroslav-dudar — APPROVED** [review]

File: airbyte-integrations/connectors/source-shopify/build.gradle
Comment (sherifnada) [inline airbyte-integrations/connectors/source-shopify/build.gradle:13]: this `dependencies` block is no longer needed

File: airbyte-integrations/connectors/source-shopify/source_shopify/source.py
Comment (sherifnada) [inline airbyte-integrations/connectors/source-shopify/source_shopify/source.py:74]: can replace with `yield from records`

File: airbyte-integrations/connectors/source-shopify/source_shopify/source.py
Comment (sherifnada) [inline airbyte-integrations/connectors/source-shopify/source_shopify/source.py:85]: are records returned in ascending order of the ID? If so we can change this to be the page size. This would save state every time a page is read which is more efficient in case of halfway failrues

File: airbyte-integrations/connectors/source-shopify/source_shopify/source.py
Comment (sherifnada) [inline airbyte-integrations/connectors/source-shopify/source_shopify/source.py:91]: type hints

File: airbyte-integrations/connectors/source-shopify/source_shopify/source.py
Comment (sherifnada) [reply airbyte-integrations/connectors/source-shopify/source_shopify/source.py:55]: @bazarnov any updates on this comment?

**Review by sherifnada — CHANGES_REQUESTED** [review]
LGTM with a few questions. I think we should be able to ship it pretty soon 🎉 

Can you paste the [definition of done template](https://github.com/airbytehq/airbyte/pull/3864) into the PR description and follow the checklist? In the future it should appear automatically but it got merged after you started this PR.

File: airbyte-integrations/connectors/source-shopify/source_shopify/source.py
Comment (bazarnov) [reply airbyte-integrations/connectors/source-shopify/source_shopify/source.py:55]: No, so far didn't get the point of how effectively to do that along with other parameters like since_id, which is doing the same operation on the data.

File: airbyte-integrations/connectors/source-shopify/source_shopify/source.py
Comment (bazarnov) [reply airbyte-integrations/connectors/source-shopify/source_shopify/source.py]: Done.

File: airbyte-integrations/connectors/source-shopify/build.gradle
Comment (bazarnov) [reply airbyte-integrations/connectors/source-shopify/build.gradle:13]: Didn't know that, fixed.

File: airbyte-integrations/connectors/source-shopify/source_shopify/source.py
Comment (bazarnov) [reply airbyte-integrations/connectors/source-shopify/source_shopify/source.py:74]: Fixed.

File: airbyte-integrations/connectors/source-shopify/source_shopify/source.py
Comment (bazarnov) [reply airbyte-integrations/connectors/source-shopify/source_shopify/source.py:85]: Bound `state_checkpoint_interval` to the `limit` constant in the superclass.

**Comment by midavadim** [comment]
We should probably delete original singer shopify connector

File: airbyte-integrations/connectors/source-shopify/source_shopify/source.py
Comment (bazarnov) [reply airbyte-integrations/connectors/source-shopify/source_shopify/source.py:91]: Set `stream_state=None`

File: airbyte-integrations/connectors/source-shopify/source_shopify/spec.json
Comment (bazarnov) [reply airbyte-integrations/connectors/source-shopify/source_shopify/spec.json:2]: changed to https://docs.airbyte.io/integrations/sources/shopify

File: airbyte-integrations/connectors/source-shopify/unit_tests/unit_test.py
Comment (bazarnov) [reply airbyte-integrations/connectors/source-shopify/unit_tests/unit_test.py:4]: Done

File: airbyte-config/init/bin/main/config/STANDARD_SOURCE_DEFINITION/9da77001-af33-4bcd-be46-6252bf9342b9.json
Comment (sherifnada) [inline airbyte-config/init/bin/main/config/STANDARD_SOURCE_DEFINITION/9da77001-af33-4bcd-be46-6252bf9342b9.json]: can you remove the svg from both this JSON and the YAML? there is currently a bug in Airbyte which will cause consumers to break when we include an icon on a new connector. A fix will be delivered but for now it should be removed

**Review by sherifnada — CHANGES_REQUESTED** [review]
two last comments

File: airbyte-config/init/bin/main/config/STANDARD_SOURCE_DEFINITION/9da77001-af33-4bcd-be46-6252bf9342b9.json
Comment (bazarnov) [reply airbyte-config/init/bin/main/config/STANDARD_SOURCE_DEFINITION/9da77001-af33-4bcd-be46-6252bf9342b9.json]: should the whole block like `"icon": "shopify.svg"` be removed, or just the `"shopify.svg"` ?

Comment (sherifnada) [reply airbyte-config/init/bin/main/config/STANDARD_SOURCE_DEFINITION/9da77001-af33-4bcd-be46-6252bf9342b9.json]: whole block

File: airbyte-integrations/connectors/source-shopify/source_shopify/spec.json
Comment (bazarnov) [reply airbyte-integrations/connectors/source-shopify/source_shopify/spec.json:10]: Working on it.

Comment (bazarnov) [reply airbyte-integrations/connectors/source-shopify/source_shopify/spec.json:10]: Done.

**Review by sherifnada — CHANGES_REQUESTED** [review]
@bazarnov can you fix the PR diff?

**Comment by bazarnov** [comment]
> @bazarnov can you fix the PR diff?

Not sure I made those changes by myself, how can I fix this? any ideas?

**Comment by bazarnov** [comment]
> > @bazarnov can you fix the PR diff?

Fixed.

File: airbyte-integrations/connectors/source-shopify/source_shopify/spec.json
Comment (sherifnada) [inline airbyte-integrations/connectors/source-shopify/source_shopify/spec.json:16]: adding a pattern to the spec allows validating it in the UI (and soon in the connector itself)
```suggestion
        "description": "The date you would like to replicate data. Format: YYYY-MM-DD.",
        "examples": ["2021-01-24"]
        "pattern": "^[0-9]{4}-[0-9]{2}-[0-9]{2}$",
```

**Review by sherifnada — COMMENTED** [review]
one question and one suggestion, almost there!

Also a couple of comments: 
1. you should edit the repo's root `.gitignore` to ignore the files which messed up the PR diff so you don't have this issue in the future
2. in the future make sure to wait for a PR approval to run the `publish` command ;)

File: airbyte-integrations/connectors/source-shopify/source_shopify/source.py
Comment (bazarnov) [reply airbyte-integrations/connectors/source-shopify/source_shopify/source.py]: Standard logic of comparison understood - edited the code.

File: airbyte-integrations/connectors/source-shopify/source_shopify/spec.json
Comment (bazarnov) [reply airbyte-integrations/connectors/source-shopify/source_shopify/spec.json:16]: Didn't know about the pattern usage, thank you for the suggestion, edited it as well.

**Review by sherifnada — APPROVED** [review]
🚢 🚢 🚢 🚢 🚢

**Comment by jacqueskpoty** [comment]
Bravo!!! for this new connector. It makes life much easier. I love the fact that it does not even use the Shopify CDK and uses the API endpoints directly. I tested it and its working perfectly for me. I'm going to add a couple of endpoints that my team needs into it as described in the following issue #3576

**Pull request merged**
